### PR TITLE
[deb] prefer mariadb over mysql

### PR DESCRIPTION
**Pull request by samueloph**
As debian Stretch, released on June this year, has mariadb as the default server instead of mysql, i thought vufind should try to install mariadb as a dependency. Please note that as there's still `virtual-mysql-server-core`as an alternate dependency, mysql-server will be accepted too.

In other words, this only change which server (mariadb|mysql) the package will ask as a dependency if any of these two aren't already installed.

Thanks

### CODE REVIEW

**Comment by demiankatz** [comment]
Thanks for sharing this!

Have you investigated the implications under Ubuntu? Since VuFind is most commonly installed on Ubuntu, does that make any difference to the decision? I'm happy to do some testing on the latest Ubuntu release when time permits if that would be helpful (though it may take me a week or two to find the time).

**Comment by samueloph** [comment]
@demiankatz This can make a difference yes, but on this case it does not.

As you can see, Ubuntu 14.04 (the oldest lts release i can investigate)[1][2][*] and newer all have mariadb-server and the virtual package virtual-mysql-server-core which guarantees that vufind can ask for mariadb-server but have mysql-server supported too.

There could be a corner case on old Ubuntu releases where the mariadb-server is an old version, like trusty [14.04] which packages mariadb-server5.5.57, but trusty does not meet the php requirements for vufind>4.0 as it ships php5.5.9, also note that although i didn't test, mariadb-server5.5.57 should be compatible with  mysql-server>=5.1.10. The following releases all ship mariadb-server>10.0.31, which i don't think would cause any mysql compatibility problem too.

On a side note, mysql 5.1 isn't supported since 2013, you may want to add mariadb and bump vufind mysql requirement to at least the last release supported now and, which is 5.5 [http://www.oracle.com/us/support/library/lifetime-support-technology-069183.pdf#page=21&zoom=auto,-266,572]. To run mysql-server 5.1 is to put yourself in a great (security) risk by now. I'm talking about this page [https://vufind.org/wiki/installation:requirements].

[1]https://packages.ubuntu.com/search?keywords=mariadb-server
[2]https://packages.ubuntu.com/trusty/virtual-mysql-server-core
*Note that i didn't check with 12.04 (precise pangolin, released April 2012), which is a release only supported for Ubuntu Advantage customers by now [https://www.ubuntu.com/info/release-end-of-life].
But anyway 12.04 doesn't meet the php requirements to run vufind>2.4 [https://web.archive.org/web/20170329003055/http://packages.ubuntu.com:80/search?keywords=php5]

**Comment by samueloph** [comment]
@demiankatz This can make a difference yes, but on this case it does not.

As you can see, Ubuntu 14.04 (the oldest lts release i can investigate)[1][2][*] and newer all have mariadb-server and the virtual package virtual-mysql-server-core which guarantees that vufind can ask for mariadb-server but have mysql-server supported too.

There could be a corner case on old Ubuntu releases where the mariadb-server is an old version, like trusty [14.04] which packages mariadb-server5.5.57, but trusty does not meet the php requirements for vufind>4.0 as it ships php5.5.9, also note that although i didn't test, mariadb-server5.5.57 should be compatible with mysql-server>=5.1.10. The following releases all ship mariadb-server>10.0.31, which i don't think would cause any mysql compatibility problem too.

On a side note, mysql 5.1 isn't supported since 2013, you may want to add mariadb and bump vufind mysql requirement to at least the last release supported now and, which is 5.5 (http://www.oracle.com/us/support/library/lifetime-support-technology-069183.pdf#page=21&zoom=auto,-266,572). To run mysql-server 5.1 is to put yourself in a great (security) risk by now. I'm talking about this page (https://vufind.org/wiki/installation:requirements).

[1]https://packages.ubuntu.com/search?keywords=mariadb-server
[2]https://packages.ubuntu.com/trusty/virtual-mysql-server-core
*Note that i didn't check with 12.04 (precise pangolin, released April 2012), which is a release only supported for Ubuntu Advantage customers by now (https://www.ubuntu.com/info/release-end-of-life).
But anyway 12.04 doesn't meet the php requirements to run vufind>2.4 (https://web.archive.org/web/20170329003055/http://packages.ubuntu.com:80/search?keywords=php5)

**Comment by demiankatz** [comment]
@samueloph, I finally had time to test this out. I tried building the package and installing it on Ubuntu 16.04. It worked; however, getting MariaDB working is more difficult than getting MySQL working due to MariaDB defaulting to locking down its root account with the unix_socket plugin. This prevents VuFind's web-based installer from working correctly.

I have added some documentation to the Ubuntu installation wiki page explaining how to change this and make things work... but I am concerned that changing the default from MySQL to MariaDB could pose a significant obstacle for novice users, increasing the number of support requests we have to field and/or discouraging some people from evaluating the software.

I'm pretty sure MariaDB does not have this problem on Fedora Linux, so this may just be an opinionated configuration found in the MariaDB package chosen by Ubuntu. Did you see this same issue on your Debian distro?

Bottom line: if we're still primarily targeting Ubuntu, I think switching to MariaDB makes installation harder without any obvious benefit (other than future-proofing in case MySQL goes away... which does count for something).

Do you have any thoughts on how we should proceed?

(Also, thanks for the note on version requirements -- I've bumped MySQL to 5.5 as you suggested).

**Comment by demiankatz** [comment]
@samueloph, I'm closing this PR due to inactivity -- I think things should continue to work as they are for now, and I remain concerned that switching the default may cause unnecessary problems. However, I am happy to resume this discussion if and when you have time, so please feel free to reopen this PR at your convenience.

**Comment by samueloph** [comment]
Hello @demiankatz, 

Yeah, i'm not having too much free time right now, i'm fine with leaving it closed, thanks.

**Comment by demiankatz** [comment]
Thanks, @samueloph, and good luck on getting caught up on things. I know the feeling of having limited free time! :-)
